### PR TITLE
Add black, ruff, and isort to development toolchain.

### PR DESCRIPTION
**Pull request by ivanklee86**
# Description
* Use ruff for linting, black for formatting, and isort to order dependencies.
* Swap out pylint (not 100%, leave comment cleanup for later).
* Update makefile and CI/CD.
* Run typing/linting on main CI branch only.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [ ] Unit tests
- [ ] Spec Tests
- [ ] Integration tests / Manual Tests

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: UnleashClient/strategies/GradualRolloutSessionId.py [hunk 1]
```diff
@@ -13,4 +13,7 @@ def apply(self, context: dict = None) -> bool:
         percentage = int(self.parameters["percentage"])
         activation_group = self.parameters["groupId"]
 
-        return percentage > 0 and normalized_hash(context["sessionId"], activation_group) <= percentage
+        return (
+            percentage > 0
+            and normalized_hash(context["sessionId"], activation_group) <= percentage
```
[the remark this answers is not in the document]
Comment (ivanklee86) [reply UnleashClient/strategies/GradualRolloutSessionId.py:18]: Sonatype & black fighting!

@sonatype-lift ignoreall

File: tests/unit_tests/test_client.py [hunk 2]
```diff
@@ -295,14 +350,18 @@ def test_uc_context_manager(unleash_client_nodestroy):
 def test_uc_not_initialized_isenabled():
     unleash_client = UnleashClient(URL, APP_NAME)
     assert not unleash_client.is_enabled("ThisFlagDoesn'tExist")
-    assert unleash_client.is_enabled("ThisFlagDoesn'tExist", fallback_function=lambda x, y: True)
+    assert unleash_client.is_enabled(
+        "ThisFlagDoesn'tExist", fallback_function=lambda x, y: True
```
[the remark this answers is not in the document]
Comment (ivanklee86) [reply tests/unit_tests/test_client.py:354]: Unit test.
@sonatype-lift ignore

File: tests/unit_tests/test_client.py [hunk 3]
```diff
@@ -271,20 +320,26 @@ def test_uc_is_enabled_with_context():
...
+    )
     responses.add(responses.POST, URL + METRICS_URL, json={}, status=202)
 
     # Create Unleash client and check initial load
     unleash_client.initialize_client()
     time.sleep(1)
     assert not unleash_client.is_enabled("ThisFlagDoesn'tExist")
-    assert unleash_client.is_enabled("ThisFlagDoesn'tExist", fallback_function=lambda x, y: True)
+    assert unleash_client.is_enabled(
+        "ThisFlagDoesn'tExist", fallback_function=lambda x, y: True
```
[the remark this answers is not in the document]
Comment (ivanklee86) [reply tests/unit_tests/test_client.py:333]: Unit test.
@sonatype-lift ignore

**Comment by sighphyre** [comment]
YES! It'd be great to have these changes in place. Looks really good so far!

**Comment by rishiraj88** [comment]
Wow! Great coverage, @coveralls !!

**Pull request merged**